### ISSUE TITLE
Volume suppression is too late / not automatic / dangerous

### ISSUE THREAD

**Issue by cyberpunkedu** [ubports/ubuntu-touch#886]
### Illustrations

- Device: Hammerhead
- Channel: 16.04/devel
- Build: 2018-09-23

### Steps to reproduce

1. Raise device volume to max
2. Plug in earbuds
3. Play a sound file (music, audio show, etc.)

### Expected behavior

Volume should automatically suppress when headphones plugged in.
OR Volume should automatically suppress during warning message.

### Actual behavior

- Warning message appears.
- Volume remains dangerously high until "cancel" is pressed.
- Ears hurt

**Comment by userj** [comment]
I would agree. It's an issue between headphones vs some external speaker. 

Upon testing plugging in the headphones vs using built-in speaker, they seem to share the same volume, whichever was set last. So as you said, it is an issue.

**Comment by userj** [comment]
It is also the case that each individual app seems to hold its own volume setting. So, even if you turn down the volume with the headphones in Podbird for example, when a call comes in it may be at max volume, or vice versa.

**Comment by garro95** [comment]
If not for individual apps, maybe at least different volumes for classes of apps (multimedia, notifications...) but still, different between headphones, built-in speaker(s), or Bluetooth devices.

**Comment by userj** [comment]
> If not for individual apps, maybe at least different volumes for classes of apps (multimedia, notifications...) but still, different between headphones, built-in speaker(s), or Bluetooth devices.

Yes, this is a continual issue. Different profiles would be important.
The volume does not react to headphones, and requires user input to change it while the headphones are plugged in. 

Try listening to audio during the day with and without headphones and notice where the issue arises. It's hard to notice unless you do both of those things regularly.

**Comment by UniversalSuperBox** [comment]
Hmm, my device changes media volume when headphones are plugged in. It also stores three different audio volumes: Media, notifications, and alarms. Each can only be set while the type of audio it affects is playing (or Notifications if no audio is playing).

**Comment by userj** [comment]
@UniversalSuperBox The device works as you described. I don't think it's a bug in function, but I think it's a bug in the design. I believe the in-call volume is also a unique store or it shares with one of the 3 you've mentioned, which is a bit confusing.

With the 3 different volume profiles, it's easy to forget if one is at an unsafe. So you might be fine talking on a call, then if you decide to listen to music after the call, you might have forgotten that the music volume was at max, and your ears get blasted *before* you have the chance to press "cancel" or "ok" on the unsafe volumes warning.

The 3 profile scenario also doesn't account for the type of speakers it's playing through. So it should work more like @garro95 suggests.

Currently, the system allows loud volumes to play before the user can decide on the outcome.

As you said, you can only adjust the volume while audio is playing. Thus, to safely adjust the volume for the headphones you must:
1. Plug in the headphones while they are not in/on your ears
2. Play the audio
3. Select weather or not you want unsafe volumes as per the popup dialogue
4. Adjust volume to where you want it
5. Put headphones on ears

Another way to listen safely is:
1. Play the audio, make a call, etc. 
2. Adjust to the appropriate level for headphones
3. Plug in headphones

I believe a safer way would work like this:

1. If headphones plugged in the volume should default/force to a safe level for all profiles.
2. Then only if the user allows "unsafe" levels, the volume would be raised again.

Additionally, I would say that the current "safe" threshold is also a bit too high, but that's subjective.

**Comment by c4pp4** [comment]
> different between headphones, built-in speaker(s), or Bluetooth devices.

Exactly. I'm using max volume for notifications from speaker. When I'm listening to music via headphones I have to turn the volume for notifications down. Sometimes I forget and my ears "thank me" for that.
